js/src/Lines.ts

```diff
@@ -670,16 +669,16 @@ export class Lines extends Mark {
     }
 
     dot: any;
-    legend_el: d3.Selection<SVGGElement, any, any, any>;
-    legend_line: d3.Line<[number, number]>;
-    legend_path_data: [number, number][];
+    legendEl: d3.Selection<any, any, any, any>;
+    legendLine: d3.Line<[number, number]>;
+    legendPathData: [number, number][];
     selector: BaseSelector;
-    selector_model: SelectorModel;
-    area: d3.Area<[number, number]>;
+    selectorModel: SelectorModel;
+    area: any;
```

Comment:
What was the problem with `d3.Area<[number, number]>`?

Reply:
This kind of error that I couldn't figure out why:
```
src/Lines.ts:433:23 - error TS2339: Property 'y0' does not exist on type 'Line<[number, number]>'.

433             this.area.y0(this.parent.plotarea_height);
                          ~~

src/Lines.ts:435:23 - error TS2339: Property 'y0' does not exist on type 'Line<[number, number]>'.

435             this.area.y0(0)
                          ~~

src/Lines.ts:437:23 - error TS2339: Property 'y0' does not exist on type 'Line<[number, number]>'.

437             this.area.y0(function(d: any) { return yScale.scale(d.y0) + yScale.offset; })
                          ~~

src/Lines.ts:473:12 - error TS2339: Property 'y1' does not exist on type 'Line<[number, number]>'.

473           .y1(function(d: any) { return yScale.scale(d.y) + yScale.offset; })
               ~~

src/Lines.ts:476:23 - error TS2339: Property 'y0' does not exist on type 'Line<[number, number]>'.

476             this.area.y0(this.parent.plotarea_height);
                          ~~

src/Lines.ts:478:23 - error TS2339: Property 'y0' does not exist on type 'Line<[number, number]>'.

478             this.area.y0(0)
                          ~~

src/Lines.ts:480:23 - error TS2339: Property 'y0' does not exist on type 'Line<[number, number]>'.

480             this.area.y0(function(d: any) { return yScale.scale(d.y0) + yScale.offset; })
                          ~~


Found 7 errors.


```
Here's a link for the original code: https://github.com/bqplot/bqplot/blob/2560b1e347d4d0f1f4a0a533c035b6957f91fd78/js/src/Lines.ts#L478

Reply:
In your error message, it complains because it thinks `this.area` is a `Line<[number, number]>`, do you know where that come from? It should think it's a `d3.Area<[number, number]>`

Reply:
lol ;/
I was copying the Line type
let's talk about this no further :see_no_evil:

Reply:
hehe thank you!